src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1078,6 +1167,10 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         use_onnx = kwargs.pop("use_onnx", None)
         load_connected_pipeline = kwargs.pop("load_connected_pipeline", False)
 
+        is_cuda_multi_device = torch.cuda.is_available() and torch.cuda.device_count() > 1
+        if device_map is not None and device_map == "auto" and not is_cuda_multi_device:
+            raise ValueError("`device_map='auto'` needs multiple CUDA devices to be present.")
```

Comment:
I think we can make it work without this condition.

Reply:
Done in b00b71d1b